Module#04/security-service/src/main/java/com/epam/esm/security/service/AuthenticationService.java

```diff
@@ -0,0 +1,235 @@
+     *
+     * @param request The AuthenticationRequest containing the username and password.
+     */
+    private void setAuthenticationToken(
+            final AuthenticationRequest request) {
+        Authentication authenticate = manager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        request.getUsername(),
+                        request.getPassword()));
+        SecurityContextHolder.getContext()
+                .setAuthentication(authenticate);
+    }
+
+    /**
+     * Retrieves the user with the given username.
+     *
+     * @param username The username of the user to find.
+     * @return The User object.
+     * @throws UserNotFoundException if the user is not found.
+     */
+    @Transactional
+    public User findUser(final String username) {
+        return userRepository.findByUsername(username)
+                .orElseThrow(() -> new UserNotFoundException(
+                        format("User not found: %s", username)));
+    }
+
+    /**
+     * Saves a new user with the provided registration details and default role.
+     *
+     * @param request The RegisterRequest containing the user registration details.
+     * @return The saved User object.
+     */
+    @Transactional
+    public User saveUserWithRole(
+            final RegisterRequest request) {
+        User user = User.builder()
+                .password(encoder.encode(request.getPassword()))
+                .username(request.getUsername())
+                .email(request.getEmail())
+                .role(getRole())
+                .build();
+        return userRepository.save(user);
+    }
+
+    /**
+     * Finds the user with the given username.
+     *
+     * @param username The username of the user to find.
+     * @return Optional containing the User object, or empty if not found.
+     */
+    @Transactional
+    public Optional<User> findByUsername(String username) {
+        return userRepository.findByUsername(username);
+    }
+
+    /**
+     * Returns the Role object representing the default role for a user.
+     *
+     * @return The default Role object.
+     */
+    private static Role getRole() {
+        return Role.builder()
+                .permission(USER)
+                .build();
+    }
+
+    /**
+     * Constructs the AuthenticationResponse object with the user details, JWT token, and access token expiration.
+     *
+     * @param user        The User object.
+     * @param jwtToken    The JWT token.
+     * @param accessToken The access token expiration time.
+     * @return The AuthenticationResponse object.
+     */
+    private AuthenticationResponse getResponse(
+            User user, String jwtToken, Long accessToken) {
+        return AuthenticationResponse.builder()
+                .username(user.getUsername())
+                .expiresAt(Timestamp.from(now()
+                        .plusMillis(accessToken)))
```

Comment:
accessToken is not understoodable name, Looks like it is object but it milis, please rename

Reply:
> accessToken is not understoodable name, Looks like it is object but it milis, please rename

> accessToken has been renamed accessTokenExpiration